File: TwoPointer/javascript/1806.js
const input = require("fs").readFileSync("/dev/stdin").toString().split("\n");
const [n, s] = input[0].split(" ").map(Number);
const arr = input[1].split(" ").map(Number);
const numSum = new Array(n + 1).fill(0);
for (let i = 1; i <= arr.length; i++) {
  numSum[i] = numSum[i - 1] + arr[i - 1];
}

let [left, right] = [0, 1];
let answer = Infinity;
while (left < n) {
  let num = numSum[right] - numSum[left];
  if (num >= s) {
    answer = Math.min(answer, right - left);
    left++;
    continue;
  }
  if (right < n) {
    right++;
    continue;
  }
  left++;
}
if (answer === Infinity) {
  answer = 0;
}

console.log(answer);

File: TwoPointer/python/1806.py
n,s = map(int,input().split())
arr = list(map(int,input().split()))
nums_sum = [0 for _ in range(len(arr)+1)]
for i in range(1,len(arr)+1):
  nums_sum[i] = nums_sum[i-1] + arr[i-1]
left,right = 0,1

infinite = 10**18*9
answer = infinite
while left<n:
  num = nums_sum[right] - nums_sum[left]
  if (num >= s):
    answer = min(answer,right-left)
    left += 1
    continue
  if right<n:
    right +=1
    continue
  left+=1
if answer == infinite:
  answer = 0
print(answer)